### ISSUE TITLE
`PYRIGHT_TYPE_CHECKING` variable

### ISSUE THREAD

**Issue by DetachHead** [KotlinIsland/basedtyping#48]
When using vscode, even when not using pyright, it still relies on it to get type information. Sometimes stuff that works in mypy  doesn't work in vscode resulting in no code completion because pyright thinks it's `Any`

In cases like that it's useful to have a `PYRIGHT_TYPE_CHECKING` variable

can use a nonsense check like `sys.version_info < (9, 9)` since pyright will treat it as `True`

**Comment by KotlinIsland** [comment]
> can use a nonsense check like `sys.version_info < (9, 9)` since pyright will treat it as `True`

[hmmm](https://mypy-play.net/?mypy=latest&python=3.10&gist=b65eed037fd5f8ffb9132d55e407934a)

**Comment by DetachHead** [comment]
more like
```py
if TYPE_CHECKING:
    PYRIGHT_TYPE_CHECKING: Literal[False] = False
    if sys.version_info != (9, 9):  # type: ignore[comparison-overlap]
        PYRIGHT_TYPE_CHECKING: Literal[True] = True  # type: ignore[no-redef]
```

**Comment by KotlinIsland** [comment]
Merged with #47
